pkg/cmd/roachtest/tests/backup.go

```diff
@@ -281,120 +284,139 @@ func registerBackup(r registry.Registry) {
+			Owner:   registry.OwnerBulkIO,
+			Cluster: KMSSpec,
+			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+				if c.Spec().Cloud == spec.GCE {
+					t.Skip("backupKMS roachtest is only configured to run on AWS", "")
+				}
 
-			// ~10GiB - which is 30Gib replicated.
-			rows := rows30GiB
-			if c.IsLocal() {
-				rows = 100
-			}
-			dest := importBankData(ctx, rows, t, c)
+				// ~10GiB - which is 30Gib replicated.
+				rows := rows30GiB
+				if c.IsLocal() {
+					rows = 100
+				}
+				dest := importBankData(ctx, rows, t, c)
 
-			conn := c.Conn(ctx, t.L(), 1)
-			m := c.NewMonitor(ctx)
-			m.Go(func(ctx context.Context) error {
-				_, err := conn.ExecContext(ctx, `
+				conn := c.Conn(ctx, t.L(), 1)
+				m := c.NewMonitor(ctx)
+				m.Go(func(ctx context.Context) error {
+					_, err := conn.ExecContext(ctx, `
 					CREATE DATABASE restoreA;
 					CREATE DATABASE restoreB;
 				`)
-				return err
-			})
-			m.Wait()
-
-			var kmsURIA, kmsURIB string
-			var err error
-			m = c.NewMonitor(ctx)
-			m.Go(func(ctx context.Context) error {
-				t.Status(`running encrypted backup`)
-				kmsURIA, err = getAWSKMSURI(KMSRegionAEnvVar, KMSKeyARNAEnvVar)
-				if err != nil {
 					return err
-				}
+				})
+				m.Wait()
+				var kmsURIA, kmsURIB string
+				var err error
+				m = c.NewMonitor(ctx)
+				m.Go(func(ctx context.Context) error {
+					switch item.kmsProvider {
+					case "AWS":
+						t.Status(`running encrypted backup with AWS KMS`)
+						kmsURIA, err = getAWSKMSURI(KMSRegionAEnvVar, KMSKeyARNAEnvVar)
+						if err != nil {
+							return err
+						}
 
-				kmsURIB, err = getAWSKMSURI(KMSRegionBEnvVar, KMSKeyARNBEnvVar)
-				if err != nil {
-					return err
-				}
+						kmsURIB, err = getAWSKMSURI(KMSRegionBEnvVar, KMSKeyARNBEnvVar)
+						if err != nil {
+							return err
+						}
+					case "GCS":
+						t.Status(`running encrypted backup with GCS KMS`)
+						kmsURIA, err = getGCSKMSURI(KMSKeyNameAEnvVar)
+						if err != nil {
+							return err
+						}
 
-				kmsOptions := fmt.Sprintf("KMS=('%s', '%s')", kmsURIA, kmsURIB)
-				_, err := conn.ExecContext(ctx,
-					`BACKUP bank.bank TO 'nodelocal://1/kmsbackup/`+dest+`' WITH `+kmsOptions)
-				return err
-			})
-			m.Wait()
+						kmsURIB, err = getGCSKMSURI(KMSKeyNameBEnvVar)
+						if err != nil {
+							return err
+						}
+					}
 
-			// Restore the encrypted BACKUP using each of KMS URI A and B separately.
-			m = c.NewMonitor(ctx)
-			m.Go(func(ctx context.Context) error {
-				t.Status(`restore using KMSURIA`)
-				if _, err := conn.ExecContext(ctx,
-					`RESTORE bank.bank FROM $1 WITH into_db=restoreA, kms=$2`,
-					`nodelocal://1/kmsbackup/`+dest, kmsURIA,
-				); err != nil {
+					kmsOptions := fmt.Sprintf("KMS=('%s', '%s')", kmsURIA, kmsURIB)
+					_, err := conn.ExecContext(ctx,
+						`BACKUP bank.bank TO 'nodelocal://1/kmsbackup/`+item.kmsProvider+`/`+dest+`' WITH `+kmsOptions)
```

Comment:
nit: can you construct the backup path once and assign it to a variable? This way the restores below can jus reuse that var, feels easier on the eyes.